crates/transpiler/src/passes/basis_translator/compose_transforms.rs

```diff
@@ -24,119 +31,177 @@ use qiskit_circuit::{
+) -> Result<IndexMap<GateIdentifier, BasisTransformOut, ahash::RandomState>, BasisTranslatorError> {
     let mut gate_param_counts: IndexMap<GateIdentifier, usize, ahash::RandomState> =
         IndexMap::default();
-    get_gates_num_params(source_dag, &mut gate_param_counts)?;
+    get_gates_num_params(source_dag, &mut gate_param_counts);
     let mut mapped_instructions: IndexMap<GateIdentifier, BasisTransformOut, ahash::RandomState> =
         IndexMap::with_hasher(ahash::RandomState::default());
 
     for (gate_name, gate_num_qubits) in source_basis.iter().cloned() {
         let num_params = gate_param_counts[&(gate_name.clone(), gate_num_qubits)];
-
-        // TODO: Follow-up by replacing with Param::ParamterExpression(Symbol),
-        // as ParameterVector is not exposed in Rust. Maybe we ought to add a function
-        // for convenient construction, if it would be useful.
-        let placeholder_params: SmallVec<[Param; 3]> = PARAMETER_VECTOR
-            .get_bound(py)
-            .call1((&gate_name, num_params))?
-            .extract()?;
+        let mut placeholder_params: SmallVec<[Param; 3]> = (0..num_params as u32)
+            .map(|idx| {
+                Param::ParameterExpression(
+                    ParameterExpression::from_symbol(Symbol::new(&gate_name, None, Some(idx)))
+                        .into(),
+                )
+            })
+            .collect();
 
         let mut dag = DAGCircuit::new();
-        // Create the mock gate and add to the circuit, use Python for this.
-        let qubits = QuantumRegister::new_owning("q".to_string(), gate_num_qubits);
-        dag.add_qreg(qubits)?;
-
-        let gate = GATE.get_bound(py).call1((
-            &gate_name,
-            gate_num_qubits,
-            placeholder_params
-                .iter()
-                .map(|x| x.clone_ref(py))
-                .collect::<SmallVec<[Param; 3]>>(),
-        ))?;
-        let gate_obj: OperationFromPython = gate.extract()?;
+        // Create the mock gate and add to the circuit, use Python if necessary.
+        let qubits = QuantumRegister::new_owning("q", gate_num_qubits);
+        dag.add_qreg(qubits).map_err(|_| {
+            BasisTranslatorError::BasisDAGCircuitError(
+                "Error while adding register to the circuit".to_string(),
+            )
+        })?;
+        let gate = if let Some(op) = name_to_packed_operation(&gate_name, gate_num_qubits) {
+            op
+        } else {
+            let extract_py = Python::with_gil(|py| -> PyResult<OperationFromPython> {
+                let gate = || -> PyResult<OperationFromPython> {
+                    GATE.get_bound(py)
+                        .call1((&gate_name, gate_num_qubits, placeholder_params.as_ref()))?
+                        .extract()
+                };
+                gate()
+            })
+            .unwrap_or_else(|_| panic!("Error creating custom gate for entry {}", gate_name));
```

Comment:
```suggestion
            .expect(format!("Error creating custom gate for entry {}", gate_name)));
```
seems more idiomatic 🙂

Reply:
This was done by clippy because `.expect` requires an `&str`. Since `format!` returns a `String` clippy automatically corrects it.

Reply:
Lol ok -- I learned something new, thanks!